data-processing/common/commands/fetch_s2_data.py

```diff
@@ -99,19 +101,29 @@ def process(self, item: ArxivId) -> Iterator[S2Metadata]:
                 data = resp.json()
                 references = []
                 for reference_data in data["references"]:
-                    authors = []
-                    for author_data in reference_data["authors"]:
-                        authors.append(Author(author_data["authorId"], author_data["name"]))
-                    reference = Reference(
-                        s2_id=reference_data["paperId"],
-                        arxivId=reference_data["arxivId"],
-                        doi=reference_data["doi"],
-                        title=reference_data["title"],
-                        authors=authors,
-                        venue=reference_data["venue"],
-                        year=reference_data["year"],
-                    )
-                    references.append(reference)
+
+                    if reference_data["paperId"]:
```

Comment:
what happens in case missing `paperId`?

Reply:
can we define a new error type complementing `S2ReferencesNotFoundException` that handles case when `if reference_data["paperId"]` is `False`?

Reply:
> can we define a new error type complementing S2ReferencesNotFoundException that handles case when if reference_data["paperId"] is False?

I'm not sure if that's going to be useful?  Esp since it's very possible that the unmatched reference turns out to be bogus.  For example, I've seen "references" that were obviously from bad extractions, where the "title" is the caption from a graph, or where the "title" is essentially a repeat of another legit reference (this happens bc different Sourced Papers can have slightly differently worded reference titles, depending on what extraction source each Sourced Paper come from).

Since the unmatched references was not a case that was a concern back when we used the legacy endpoint, I figured that we probably don't need to care about them here anyway.

Reply:
I see a case for raising an exception, in that it lets the caller decide if they care.

But I also see the case for not raising an exception, as it means we have to further digest the code we're changing (which we haven't changed in awhile). We have to figure out where the associated code is called, and handle the exception we're adding (to preserve the current behavior).

I favor the latter because it's less work. But I'd consider adding a log message whenever we skip a reference because of this, so that we know something is being filtered out at runtime. That way if someone is investigating something that's related, the log message might help them deduce why quickly. I'd also make a note explaining this. You could even link to this discussion thread.

FWIW it's surprising to me that the API returns references that aren't in-corpus. But that's probably a lot more context that plays into that decision. I wonder if we could add a parameter to the API that lets people filter the response to only include in-corpus things `?discard_nonexistant=1` or something.

Reply:
> FWIW it's surprising to me that the API returns references that aren't in-corpus. But that's probably a lot more context that plays into that decision. I wonder if we could add a parameter to the API that lets people filter the response to only include in-corpus things ?discard_nonexistant=1 or something.

interesting, yeah IDK the new Public API endpoint does that, I think we didn't realize the difference at the time.  But yeah, I will check with that team and see if it's soemthing they're interested in

Reply:
> I favor the latter because it's less work. But I'd consider adding a log message whenever we skip a reference because of this, so that we know something is being filtered out at runtime. That way if someone is investigating something that's related, the log message might help them deduce why quickly. I'd also make a note explaining this. You could even link to this discussion thread.

Oh yeah... a log would be useful, will add!

Reply:
@kyleclo do you mind adding this [here as a feature request from S2 research](https://docs.google.com/document/d/1Qtc-vdMv5RA6RRXtyriegLg6u21OiSgz7k8osIUykJU/edit#bookmark=id.f227a6gjy21k) so I can track and prioritize it as needed for the following quarters? I already added a placeholder but it'd be great if someone who experienced the user pain around this can elaborate.